File: 2024/day08/day08_test.go
package day08

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestPart1(t *testing.T) {
	assert.Equal(t, 14, Part1("input_example_1"))
	assert.Equal(t, 299, Part1("input"))
}

func TestPart2(t *testing.T) {
	assert.Equal(t, 34, Part2("input_example_1"))
	assert.Equal(t, 1032, Part2("input"))
}

File: 2025/test/day_08_test.exs
defmodule Day07Test do
  use ExUnit.Case

  import TestHelper

  test "part1" do
    assert Day08.part_1(input("day_08_example_1"), 10) == 40
    assert Day08.part_1(input("day_08"), 1000) == 75680
  end

  test "part2" do
    assert Day08.part_2(input("day_08_example_1")) == 25272
    assert Day08.part_2(input("day_08")) == 8_995_844_880
  end
end
